pydivide/read.py

```diff
@@ -66,15 +100,30 @@ def read(input_time, instruments=None, insitu_only=False):
             input_time = input_time + ' 00:00:00'
         date1 = parse(input_time)
         date2 = date1 + timedelta(days=1)
-        
+
     date1_unix = calendar.timegm(date1.timetuple())
     date2_unix = calendar.timegm(date2.timetuple())
-    
-    filenames = get_latest_files_from_date_range(date1, date2)
-    iuvs_filenames = get_latest_iuvs_files_from_date_range(date1, date2)
-    if not filenames and not iuvs_filenames:
-        print("No file found for this date range.")
-        return
+
+    # Grab insitu and iuvs files for the specified/created date ranges
+    date_range_filenames = get_latest_files_from_date_range(date1, date2)
+    date_range_iuvs_filenames = get_latest_iuvs_files_from_date_range(date1, date2)
+
+    if not date_range_filenames and not date_range_iuvs_filenames:
+        if not filenames and not iuvs_filenames:
+            print("No files found for the input date range, and no specific filenames were given. Exiting.")
+            return
+
+    # Going to look for files between time frames, but as we might have already specified
+    # certain files to load in, we don't want to load them in 2x... so doing a check for that here
+    if date_range_filenames:
+        for file in date_range_filenames:
+            if file in filenames:
+                continue
+
+    if date_range_iuvs_filenames:
+        for file in date_range_iuvs_filenames:
+            if file in iuvs_filenames:
+                continue
     
     kp_insitu = []
     if filenames:
```

Comment:
It seems like this variable won't get set anymore if only the input  time is given

Reply:
mmm, good point, investigating

Reply:
I believe I circumvented this issue with what I changed above.